### PR TITLE
[13.0][ADD] helpdesk_mgmtsystem_nonconformity: New addon

### PR DESCRIPTION
**Pull request by victoralmau**
This module allows you to create a nonconformity from a helpdesk ticket, and keeps it linked.

Locked by:
- [x] `mgmtsystem_nonconformity` https://github.com/OCA/management-system/pull/401

Please @chienandalu and @pedrobaeza can you review it?

@Tecnativa TT33874

### CODE REVIEW

File: helpdesk_mgmtsystem_nonconformity/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Helpdesk Management - Nonconformity",
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/__manifest__.py:4]:
```suggestion
    "name": "Helpdesk Management - Nonconformity",
    "summary": "Links helpdesk tickets with nonconformities"
```

File: helpdesk_mgmtsystem_nonconformity/demo/helpdesk_ticket_stage.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/demo/helpdesk_ticket_stage.xml]: These stages are data in main module, not demo, so switch it to data as well.

File: helpdesk_mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class MgmtsystemNonconformity(models.Model):
+    _inherit = "mgmtsystem.nonconformity"
+
+    ticket_id = fields.Many2one(comodel_name="helpdesk.ticket", string="Ticket",)
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py]: This should be a one2many, the inverse of the other m2o, as you can link several helpdesk tickets (several notifications of the same problem), to the same non conformity.

File: helpdesk_mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_stage.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MgmtsystemNonconformityStage(models.Model):
+    _inherit = "mgmtsystem.nonconformity.stage"
+
+    helpdesk_stage_ids = fields.One2many(
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_stage.py]: Why a one2many? Maybe you have to restrict the possibility of linking to more than one stages. If not, the stage auto-switch may be not correct.

File: helpdesk_mgmtsystem_nonconformity/models/helpdesk_ticket_stage.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+from odoo import fields, models
+
+
+class HelpdeskTicketStage(models.Model):
+    _inherit = "helpdesk.ticket.stage"
+
+    nonconformity_stage_id = fields.Many2one(
+        comodel_name="mgmtsystem.nonconformity.stage", string="Nonconformity Stage",
+    )
+    _sql_constraints = [
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/models/helpdesk_ticket_stage.py]: Remove this constraint, as you can have several stages that leads to the same non conformity stage.

File: helpdesk_mgmtsystem_nonconformity/views/helpdesk_ticket_view.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            </field>
+        </field>
+    </record>
+    <record id="ticket_view_form" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.form</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <field name="nonconformity_id" invisible="1" />
```
Comment (pedrobaeza) [inline helpdesk_mgmtsystem_nonconformity/views/helpdesk_ticket_view.xml]: Please add this field to be editable for manual linking without creating a new non conformity.

**Review by pedrobaeza — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Pull request merged**